Sources/SwiftBSON/ExtendedJSONDecoder.swift

```diff
@@ -17,8 +17,15 @@ public class ExtendedJSONDecoder {
     }()
 
     /// A set of all the possible extendedJSON wrapper keys.
+    /// This does not include the legacy extended JSON wrapper keys.
```

Comment:
Since we allow things like "$type" to be in a regular document (e.g. as part of a query operator), the legacy wrapper keys can't be used to determine errors like we do with the reserved ones, so we need to separate them out.